pkg/slogctx/slogctx.go

```diff
@@ -0,0 +1,43 @@
+// Slogctx defines contextual slog utilities, that allow for tying a slog logger with each context.
+// Context sources should usually start with a `FromContext(ctx, "logger", "my-module-here")` such that all outputs have a logger attribute.
+package slogctx
+
+import (
+	"context"
+	"log/slog"
+)
+
+// key is defined to be a separate type for Context. This is recommended practice, per Context.Value.
+type key int
+
+// The loggerKey is the default key in the context. It is a separate type from any other, so no value is necessary. This is recommended practice, per Context.Value.
+var loggerKey key
+
+// With returns a Logger that includes the given attributes in each output operation.
+// Arguments are converted to attributes as if by [Logger.Log].
+// The attached context also includes this new logger.
+func With(ctx context.Context, logger *slog.Logger, values ...any) (context.Context, *slog.Logger) {
```

Comment:
Similar to the other comment. I find it a bit confusing that it does 3 things instead of being single-purpose: 
1. Set values in the context.
2. Set values in the logger. 
3. Set the logger in the context. 


I would keep only 1) in this function. If you want to do 2) and 3), you can still do it in these 2 ways: 

```go
logger := slogctx.From(ctx)
logger.With("request", "GET")
ctx = slogctx.To(logger)
```
Or
```
ctx = slogctx.With(ctx, "request", "GET")
logger := slogctx.From(ctx)
```

Reply:
I think this and the other comment go well together, so will answer them both in one:

In the vast majority of cases, we want to both get the logger, add a little more value to it, then finally set it to the context for further use. I think saving a couple lines on every call with the small risk we do a little bit of unnecessary work is quite valuable here, as it helps propagating values further downstream.
In the few cases where we do only want 1 of the values out, it is solved by simply discarding the other.

Reply:
I see. 

I believe you are seeing that a bigger need for this because of the other pattern of adding `logger` and `component` fields in every place. See my comment on the `AddSource` and also the one about not passing too many values downstream. 

Let's discuss again after you look at those and the possible updates :)